File: backend/models/property/cashLedger.model.js
module.exports = (sequelize, DataTypes) => {
    const CashLedger = sequelize.define('cash_ledger', {
        outlet_id: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        txn_date: {
            type: DataTypes.DATEONLY,
            allowNull: false
        },
        transaction_type: {
            type: DataTypes.STRING(50),
            allowNull: false
        },
        reference_type: DataTypes.STRING(50),
        reference_id: DataTypes.INTEGER,
        reference_no: DataTypes.STRING(50),
        party_name: DataTypes.STRING(150),
        payment_method: DataTypes.STRING(20),
        amount_in: {
            type: DataTypes.DECIMAL(12, 2),
            defaultValue: 0
        },
        amount_out: {
            type: DataTypes.DECIMAL(12, 2),
            defaultValue: 0
        },
        adjustment_amount: {
            type: DataTypes.DECIMAL(12, 2),
            defaultValue: 0
        },
        balance: {
            type: DataTypes.DECIMAL(12, 2),
            defaultValue: 0
        },
        notes: DataTypes.TEXT,
        created_by: DataTypes.INTEGER
    }, {
        tableName: 'cash_ledger',
        timestamps: true,
        createdAt: 'created_at',
        updatedAt: false
    });

    return CashLedger;
};

File: lib/models/reports/cash_ledger_model.dart
class CashLedgerEntry {
  final DateTime txnDate;
  final String transactionType;
  final String referenceNo;
  final String partyName;
  final String paymentMethod;
  final double amountIn;
  final double amountOut;
  final double balance;
  final String notes;

  CashLedgerEntry({
    required this.txnDate,
    required this.transactionType,
    required this.referenceNo,
    required this.partyName,
    required this.paymentMethod,
    required this.amountIn,
    required this.amountOut,
    required this.balance,
    required this.notes,
  });

  factory CashLedgerEntry.fromJson(Map<String, dynamic> json) {
    return CashLedgerEntry(
      txnDate: DateTime.parse(json['txn_date']),
      transactionType: (json['transaction_type'] ?? '').toString(),
      referenceNo: (json['reference_no'] ?? '').toString(),
      partyName: (json['party_name'] ?? '').toString(),
      paymentMethod: (json['payment_method'] ?? '').toString(),
      amountIn: double.tryParse((json['amount_in'] ?? 0).toString()) ?? 0,
      amountOut: double.tryParse((json['amount_out'] ?? 0).toString()) ?? 0,
      balance: double.tryParse((json['balance'] ?? 0).toString()) ?? 0,
      notes: (json['notes'] ?? '').toString(),
    );
  }
}
